### PR TITLE
fix(security) fix sql injection on getServiceXML

### PR DESCRIPTION
**Pull request by loiclau**
## Description

sql injection

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x
- [ ] 18.10.x
- [x] 19.04.x
- [x] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/configuration/configObject/service_dependency/getServiceXml.php [hunk 1]
```diff
@@ -60,7 +60,7 @@
  */
 $centreon = $_SESSION['centreon'];
 $acl = $centreon->user->access;
-$hostId = $_POST['host_id'];
+$hostId = filter_var($_POST['host_id'], FILTER_SANITIZE_NUMBER_INT);
```
Comment (sc979) [inline www/include/configuration/configObject/service_dependency/getServiceXml.php:63]:
```suggestion
$hostId = filter_var($_POST['host_id'], FILTER_VALIDATE_INT);
```

File: www/include/configuration/configObject/service_dependency/getServiceXml.php [hunk 2]
```diff
@@ -107,7 +107,7 @@
 					SELECT s.service_id, s.service_description, h.host_name, h.host_id 
 					FROM service s, host h, host_service_relation hsr 
 					WHERE hsr.hostgroup_hg_id IS NULL 
-                                        AND h.host_id = '" . $db->escape($hostId). "' 
+                                        AND h.host_id = " . (int)$hostId . "
```
Comment (sc979) [inline www/include/configuration/configObject/service_dependency/getServiceXml.php:110]: shouldn't you bind the param ?

File: www/include/configuration/configObject/service_dependency/getServiceXml.php [hunk 3]
```diff
@@ -117,8 +117,8 @@
 					WHERE hsr.host_host_id IS NULL 
                                         AND hsr.hostgroup_hg_id IN (SELECT hostgroup_hg_id 
                                                                     FROM hostgroup_relation 
-                                                                    WHERE host_host_id = '" . $db->escape($hostId). "') 
-                                        AND h.host_id = '" . $db->escape($hostId). "' 
+                                                                    WHERE host_host_id = " . (int)$hostId . ") 
+                                        AND h.host_id = " . (int)$hostId . "
```
Comment (sc979) [inline www/include/configuration/configObject/service_dependency/getServiceXml.php:121]: shouldn't you bind the param ?

File: www/include/configuration/configObject/service_dependency/getServiceXml.php [hunk 4]
```diff
@@ -117,8 +117,8 @@
 					WHERE hsr.host_host_id IS NULL 
                                         AND hsr.hostgroup_hg_id IN (SELECT hostgroup_hg_id 
                                                                     FROM hostgroup_relation 
-                                                                    WHERE host_host_id = '" . $db->escape($hostId). "') 
-                                        AND h.host_id = '" . $db->escape($hostId). "' 
+                                                                    WHERE host_host_id = " . (int)$hostId . ")
```
Comment (sc979) [inline www/include/configuration/configObject/service_dependency/getServiceXml.php:120]: odd indentation

File: www/include/configuration/configObject/service_dependency/getServiceXml.php (hunk 2 above)
Comment (sc979) [inline www/include/configuration/configObject/service_dependency/getServiceXml.php:110]: you could avoid sending a useless query by checking if hostId is false on line 87 -> 
https://github.com/centreon/centreon/pull/8758/files#diff-2301f8bfcef29a6676ab357c678d03d8R87

File: www/include/configuration/configObject/servicegroup/getServiceXml.php [hunk 5]
```diff
@@ -49,7 +49,7 @@
 $centreon = $_SESSION['centreon'];
 $db = new CentreonDB();
 
-$hostId = $_POST['host_id'];
+$hostId = filter_var($_POST['host_id'], FILTER_SANITIZE_NUMBER_INT);
```
Comment (sc979) [inline www/include/configuration/configObject/servicegroup/getServiceXml.php:52]:
```suggestion
$hostId = filter_var($_POST['host_id'], FILTER_VALIDATE_INT);
```

File: www/include/configuration/configObject/servicegroup/getServiceXml.php [hunk 6]
```diff
@@ -64,7 +64,7 @@
     		  WHERE s.service_id = hsr.service_service_id
     		  AND hsr.host_host_id = h.host_id ";
     if ($hostId) {
-        $query .= " AND h.host_id = " . $db->escape($hostId);
+        $query .= " AND h.host_id = " . (int)$hostId;
```
Comment (sc979) [inline www/include/configuration/configObject/servicegroup/getServiceXml.php:67]: you could avoid sending a useless query by checking if hostId is false on line 56 ->
https://github.com/centreon/centreon/pull/8758/files#diff-d4cc75f9c7c6ad77d67f6868d48c2574R56

Comment (sc979) [inline www/include/configuration/configObject/servicegroup/getServiceXml.php:67]: could you bind the param ?

File: www/include/configuration/configObject/service_dependency/getServiceXml.php [hunk 7]
```diff
@@ -117,17 +117,19 @@
...
                                         AND s.service_register = '1' 
                                 ) AS res $aclFrom $aclCond
 				ORDER BY res.host_name, res.service_description";
     }
-    $res = $db->query($query);
-    while ($row = $res->fetchRow()) {
+    $stmt = $db->query($query);
+    $stmt->bindValue(':hostId', $hostId, PDO::PARAM_INT);
+    $stmt->execute();
+    while ($row = $stmt->fetchRow()) {
```
Comment (callapa) [inline www/include/configuration/configObject/service_dependency/getServiceXml.php:130]:
```suggestion
    while ($row = $stmt->fetch(\PDO::FETCH_ASSOC)) {
```

File: www/include/configuration/configObject/servicegroup/getServiceXml.php [hunk 8]
```diff
@@ -49,32 +49,32 @@
 $centreon = $_SESSION['centreon'];
 $db = new CentreonDB();
 
-$hostId = $_POST['host_id'];
+$hostId = filter_var($_POST['host_id'], FILTER_VALIDATE_INT);
 $acl = $centreon->user->access;
 $xml = new CentreonXML();
 $xml->startElement("response");
-if ($hostId != "") {
+if ($hostId) {
```
Comment (callapa) [inline www/include/configuration/configObject/servicegroup/getServiceXml.php:56]: Logically incorrect, what's happen if id = 0 ?
```suggestion
if ($hostId !== false) {
```

Comment (sc979) [reply www/include/configuration/configObject/servicegroup/getServiceXml.php:56]: $hostId !== false && $hostId > 0

File: www/include/configuration/configObject/service_dependency/getServiceXml.php (hunk 7 above)
Comment (sc979) [reply www/include/configuration/configObject/service_dependency/getServiceXml.php:130]: @callapa -> as discussed, the fetchRow() method is called in a while loop, it won't work with a FETCH_ASSOC

**Comment by loiclau** [comment]
finally, vulnerable files are not use (old select replace by select2)

**Pull request merged**